csharp/src/Ice/Communicator.cs

```diff
@@ -1015,48 +1015,38 @@ public void Dispose()
             _state = StateActive;
             _compactIdResolver = compactIdResolver;
             _dispatcher = dispatcher;
-            _logger = logger!;
+            _logger = logger ?? Util.getProcessLogger();
             _observer = observer;
             _threadStart = threadStart;
             _threadStop = threadStop;
             _typeIdNamespaces = typeIdNamespaces ?? new string[] { "Ice.TypeId" };
 
-            if (properties != null)
-            {
-                foreach (var entry in properties)
-                {
-                    _properties[entry.Key] = new PropertyValue(entry.Value, false);
-                }
-            }
+            properties ??= new Dictionary<string, string>();
```

Comment:
Shouldn't we clone the properties argument, to avoid modifications to it? We don't want the appSettings or args added to the properties dictionary that the user pass as an argument.

```
properties = properties == null  ? new Dictionary<string, string>() : new Dictionary<string, string>(properties);
```

Reply:
Yes fixed